File: PruebaPacientes/PruebaPacientes/PruebaPacientes/AntecedenteEpidemiologicos1.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Modelo
{
    public class AntecedenteEpidemiologicos1
    {
        public int Id                                                           { get; set; }
       
        public TipoDeOcupacionDelPaciente TipoDeOcupacionDelPaciente            { get; set; }
        public string OtroTipoDeOcupacion                                       { get; set; }

        public bool TieneVacunaAntiGripal                                       { get; set; }
        public DateTime FechaDeVacunacion                                       { get; set; }
        public DateTime FechaDeVencimientoDeLaVacuna                            { get; set; }

        public bool EstuvoEnUnLugarDeContagioFueraDelPaisPrevioASintomas        { get; set; }
        public string PaisEnQueEstuvo                                           { get; set; }
        public string CiudadDelPaisEnQueEstuvo                                  { get; set; }
        public DateTime FechaDeLlegadaAlPaisEnQueEstuvo                         { get; set; }
        public DateTime FechaDeSalidaDelPaisEnQueEstuvo                         { get; set; }
        public TipoDeTransporteUtilizado TipoDeTransporteUtilizado              { get; set; }
        public DateTime FechaDeIngresoAlPais                                    { get; set; }
        public string CompañiaDeTransporte                                      { get; set; }

        public bool ResidioEnLugarAjenoASuDomicilioDentroDelPaisPrevioASintomas { get; set; }
        public Domicilio DomicilioAlejoEnElQueResidio                           { get; set; }
        public DateTime FechaDeLlegadaADomicilioAjeno                           { get; set; }
        public DateTime FechaDeSalidaDelDomicilioAjeno                          { get; set; }
        
        //public int FormularioParaAnalisisId {get; set; }
        //public FormularioParaAnalisis FormularioParaAnalisis { get; set; }


    }
}

File: Frontend/api/models/AntecedenteEpidemiologicos1.js
/**
 * AntecedenteEpidemoliogicos1.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {

    TipoDeOcupacionDelPaciente:{ //Modelo Uno-Muchos
      model: 'TipoDeOcupacionDelPaciente'
    },
    OtroTipoDeOcupacion:{
      type: 'string'
    },
    TieneVacunaAntiGripal:{
      type: 'boolean',
      defaultsTo: false
    },
    FechaDeVacunacion:{
      type: 'string'
    },
    FechaDeVencimientoDeLaVacuna:{
      type: 'string'
    },
    EstuvoEnUnLugarDeContagioFueraDelPaisPrevioASintomas:{
      type: 'boolean',
      defaultsTo: false
    },
    PaisEnQueEstuvo:{
      type: 'string'
    },
    CiudadDelPaisEnQueEstuvo:{
      type: 'string'
    },
    FechaDeLlegadaAlPaisEnQueEstuvo:{
      type: 'string'
    },
    FechaDeSalidaAlPaisEnQueEstuvo:{
      type: 'string'
    },
    TipoDeTransporteUtilizado:{ //Modelo Uno-Muchos
      model: 'TipoDeTransporteUtilizado'
    },
    FechaDeIngresoAlPais:{
      type: 'string'
    },
    CompaniaDeTransporte:{
      type: 'string'
    },
    ResidioEnLugarAjenoASuDomicilioDentroDelPaisPrevioASintomas:{
      type: 'boolean',
      defaultsTo: false
    },
    DomicilioAjenoEnElQueResidio:{ //Modelo domicilio Uno-Uno ***CORREGIR
      /*collection: 'Domicilio',
      via:'owner'*/
      type:'string'
    },
    FechaDeLlegadaADomicilioAjeno:{
      type: 'string'
    },
    FechaDeSalidaDelDomicilioAjeno:{
      type: 'string'
    },
    //Relacion Uno-Uno
    owner:{
      model:'FormularioParaAnalisis',
      unique: true
    }

  },

};

